### PR TITLE
Add runtime params --long-dist-demand-forecast and --freight-matrix-path

### PR DESCRIPTION
**Pull request by zptro**
Added:
`--long-dist-demand-forecast`: "If `"calc"`, runs assigment with free-flow speed and calculates demand for long-distance trips. If `"base"`, takes long-distance trips from base matrices. If path, takes long-distance trips from that path."
`--freight-matrix-path`: "If specified, take freight demand matrices from path."

Removed:
`--free-flow-assignment`: This is replaced by `--long-dist-demand-forecast`

FYI: @SelinaMT @Konnde

### CODE REVIEW

File: Scripts/modelsystem.py [hunk 1]
```diff
@@ -168,52 +178,50 @@ def _add_internal_demand(self, previous_iter_impedance, is_last_iteration):
...
-                        self.dtm.add_demand(demand)
+        # If we want to assign all trips with traffic congestion,
+        # then add long-distance trips as background demand
+        zone_numbers = self.ass_model.zone_numbers
+        car_matrices = {}
+        if long_dist_matrices is not None:
+            # Try getting long-distance trips from separate file
+            with long_dist_matrices.open(
+                    "demand", "vrk", zone_numbers,
+                    self.mapping, long_dist_classes) as mtx:
```
Comment (zptro) [inline Scripts/modelsystem.py:192]: This is the only functional change here, but unfortunately the change of indentation hides it.

**Review by attesn — CHANGES_REQUESTED** [review]
Test run gave error below when using command line and dev-config.json settings.

Other minor issue for discussion is misspelled paths. Using erroneous path led to error message `OSError: ``demand_vrk.omx`` does not exist`. Should correct paths be validated?

File: Scripts/lem.py [hunk 2]
```diff
@@ -15,9 +15,13 @@
 
 
 def main(args):
+    calculate_long_dist_demand = args.long_dist_demand_forecast == "calc"
+    long_dist_matrices_path = (None
+        if args.long_dist_demand_forecast in ("calc", "base")
+        else Path(args.long_dist_demand_forecast))
```
[the remark this answers is not in the document]
Comment (zptro) [reply Scripts/lem.py:21]: Thanks for pointing this out. I was about to update dev-config.json, but then I somehow forgot. Now it is fixed.

**Comment by zptro** [comment]
> Test run gave error below when using command line and dev-config.json settings.
> 
> Other minor issue for discussion is misspelled paths. Using erroneous path led to error message `` OSError: ``demand_vrk.omx`` does not exist ``. Should correct paths be validated?

Validating paths is a tricky thing here, because that is normally done in `lem_validate_inputfiles.py` before all the model runs, but in this case the long-distance matrices (or even the directory where to find them) may not exist yet (because the long-distance demand calculation is one of the model runs that are supposed to be run).

**Review by attesn — APPROVED** [review]

**Pull request merged**